src/cfn/listStacks.ts

```diff
@@ -0,0 +1,123 @@
+        return _.every(tagsFilter, ([k, v]) => tags[k] === v);
+      });
+    }
+    if (jmespathFilter) {
+      predicates.push((stack: aws.CloudFormation.Stack) => {
+        const jmespathResult = jmespath.search(stack, jmespathFilter);
+        logger.debug(`jmespath filtered: ${stack.StackId} jmespathResult=${jmespathResult}`);
+        return !!jmespathResult;
+      });
+    }
+    const combinedPredicate: (stack: aws.CloudFormation.Stack) => boolean = _.overEvery(predicates);
+    filteredStacks = _.filter(stacks, combinedPredicate);
+  }
+  else {
+    filteredStacks = stacks;
+  }
+  if (query) {
+    // TODO consider adding in .Resources
+    console.log(JSON.stringify(jmespath.search({Stacks: filteredStacks}, query), null, ' '));
+    return;
+  }
+  else {
+    console.log(cli.blackBright(`Creation/Update Time, Status, Name${showTags ? ', Tags' : ''}`));
+    for (const stack of filteredStacks) {
+      const tags = _.fromPairs(_.map(stack.Tags, (tag) => [tag.Key, tag.Value]));
+      const lifecyle: string | undefined = tags.lifetime;
+      let lifecyleIcon: string = '';
+      if (stack.EnableTerminationProtection || lifecyle === 'protected') {
+        // NOTE stack.EnableTerminationProtection is undefined for the
+        // time-being until an upstream bug is fix by AWS
+        lifecyleIcon = '🔒 ';
+      }
+      else if (lifecyle === 'long') {
+        lifecyleIcon = '∞ ';
+      }
+      else if (lifecyle === 'short') {
+        lifecyleIcon = '♺ ';
+      }
+      const baseStackName = stack.StackName.startsWith('StackSet-')
+        ? `${cli.blackBright(stack.StackName)} ${tags.StackSetName || stack.Description || 'Unknown stack set instance'}`
+        : stack.StackName;
+      let stackName: string;
+      if (stack.StackName.includes('production') || tags.environment === 'production') {
+        stackName = cli.red(baseStackName);
+      }
+      else if (stack.StackName.includes('integration') || tags.environment === 'integration') {
+        stackName = cli.xterm(75)(baseStackName);
+      }
+      else if (stack.StackName.includes('development') || tags.environment === 'development') {
+        stackName = cli.xterm(194)(baseStackName);
+      }
+      else {
+        stackName = baseStackName;
+      }
+      process.stdout.write(
+        sprintf('%s %s %s %s\n',
+          formatTimestamp(sprintf(`%${timePadding}s`, renderTimestamp(def(stack.CreationTime, stack.LastUpdatedTime)))),
+          colorizeResourceStatus(stack.StackStatus, statusPadding),
+          cli.blackBright(lifecyleIcon) + stackName,
+          showTags ? cli.blackBright(prettyFormatTags(stack.Tags)) : ''));
+      if (stack.StackStatus.indexOf('FAILED') > -1 && !_.isEmpty(stack.StackStatusReason)) {
+        console.log('  ', cli.blackBright(stack.StackStatusReason));
+      }
+    }
+  }
+}
+
+export async function listStacksMain(argv: GenericCLIArguments): Promise<number> {
```

Comment:
It seems to me that the CLI-related stuff should not be within the `cfn` directory – what do you think?

Reply:
I think it should be in there but not littered along side the util code.